firefox-ios/Client/Frontend/Browser/TabScrollController/TabScrollHandler.swift

```diff
@@ -7,8 +7,21 @@ import SnapKit
 import Shared
 import Common
 
-final class TabScrollController: NSObject,
+protocol TabScrollHandlerDelegate: AnyObject {
+    func startAnimatingToolbar(state: ToolbarState)
+    func showToolbar()
+    func hideToolbar()
+}
```

Comment:
nit: Starting form swift 5.10 we can use nesting protocols, so we can add the bellow code in `TabScrollHandler` class.
```
protocol Delegate: AnyObject {
        func startAnimatingToolbar(state: ToolbarState)
        func showToolbar()
        func hideToolbar()
    }
```